service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/WorkspaceCloneUtils.java

```diff
@@ -122,6 +131,21 @@ private static ReferencedResource buildDestinationBigQueryDatasetReference(
     return resultBuilder.build();
   }
 
+  private static ReferencedResource buildDestinationBigQueryDataTableReference(
+      ReferencedBigQueryDataTableResource sourceBigQueryResource,
+      UUID destinationWorkspaceId,
+      @Nullable String name,
+      @Nullable String description) {
+    // keep projectId and dataset name the same since they are for the referent
+    final ReferencedBigQueryDataTableResource.Builder resultBuilder =
+        sourceBigQueryResource.toBuilder()
+            .workspaceId(destinationWorkspaceId)
+            .resourceId(UUID.randomUUID());
+    Optional.ofNullable(name).ifPresent(resultBuilder::name);
```

Comment:
I introduced this style, but I don't think we need to guard against null here. It's really only necessary if the builder throws on null.

Reply:
it does throw at line 146 if name or description is null because validate() throws MissingRequiredFieldException when name is null or empty.